torch/testing/_internal/common_utils.py

```diff
@@ -459,29 +459,28 @@ def __init__(self, arg_str, arg_values, name_fn=None):
         self.arg_values = arg_values
         self.name_fn = name_fn
 
-    def _formatted_str_repr(self, name, value):
+    def _formatted_str_repr(self, idx, name, value):
         """ Returns a string representation for the given arg that is suitable for use in test function names. """
         if isinstance(value, torch.dtype):
             return dtype_name(value)
         elif isinstance(value, torch.device):
             return str(value)
         # Can't use isinstance as it would cause a circular import
-        elif value.__class__.__name__ == 'OpInfo' or value.__class__.__name__ == 'ModuleInfo':
+        elif type(value).__name__ in {'OpInfo', 'ModuleInfo'}:
             return value.formatted_name
         else:
-            # Include name and value separated by underscore.
-            return f"{name}_{str(value).replace('.', '_')}"
+            return f"{name}{idx}"
```

Comment:
This is the actual fix. Everything else are just modifications to get access to the index here.